src/object/object_primitive.c

```diff
@@ -17445,9 +17450,16 @@ mr_data_lengthval_json (DB_VALUE * value, int disk)
       return tp_Json.size;
     }
 
-  if (value->data.json.json_body != NULL)
+  if (value->data.json.document != NULL)
```

Comment:
mr_data_lengthval_json, mr_data_writeval_json and mr_data_readval_json should call mr_data_lengthmem_json, mr_data_writemem_json and mr_data_readmem_json respectively on ```&value->data.json```.